query/src/main/scala/filodb/query/exec/SelectRawPartitionsExec.scala

```diff
@@ -46,80 +48,151 @@ object SelectRawPartitionsExec extends  {
-        colIds ++ dataSchema.colIDs(dataSchema.data.valueColName).get
-      } else {
-        // need to select column based on range function for gauge schema
-        val colNames = rangeVectorTransformers.find(_.isInstanceOf[PeriodicSamplesMapper]).map { p =>
-          RangeFunction.downsampleColsFromRangeFunction(dataSchema, p.asInstanceOf[PeriodicSamplesMapper].functionId)
-        }.getOrElse(Seq(dataSchema.data.valueColName))
-        colIds ++ dataSchema.colIDs(colNames: _*).get
-      }
-    }
+  def doExecute(source: ChunkSource,
+                queryConfig: QueryConfig)
+               (implicit sched: Scheduler,
+                timeout: FiniteDuration): ExecResult = {
+    Query.qLogger.debug(s"queryId=$id on dataset=$datasetRef shard=${lookupRes.map(_.shard).getOrElse("")}" +
+      s"schema=${dataSchema.map(_.name)} is configured to use columnIDs=$colIds")
+    val rvs = dataSchema.map { sch =>
+      source.rangeVectors(datasetRef, lookupRes.get, colIds, sch, filterSchemas)
+    }.getOrElse(Observable.empty)
+    ExecResult(rvs, Task.eval(schemaOfDoExecute()))
   }
 
-  protected def doExecute(source: ChunkSource,
-                          queryConfig: QueryConfig)
-                         (implicit sched: Scheduler,
-                          timeout: FiniteDuration): Observable[RangeVector] = {
-    require(!colIds.contains(0),
-      "User selected columns should not include timestamp (row-key); it will be auto-prepended")
+  protected def args: String = s"dataset=$dataset, shard=${lookupRes.map(_.shard).getOrElse(-1)}, " +
+                               s"schema=${dataSchema.map(_.name)}, filterSchemas=$filterSchemas, " +
+                               s"chunkMethod=${lookupRes.map(_.chunkMethod).getOrElse("")}, colIDs=$colIds"
+}
 
-    // Optimize/adjust transformers for downsampling queries.
-    replaceTransformers(optimizeForDownsample(dataSchema, rangeVectorTransformers))
+/**
+  * ExecPlan to select raw data from partitions that the given filter resolves to,
+  * in the given shard, for the given row key range.  Schema-agnostic - discovers the schema and
+  * creates an inner SelectRawPartitionsExec with the right column IDs and other details depending on schema.
+  * @param schema an optional schema to filter partitions using
+  */
+final case class MultiSchemaPartitionsExec(id: String,
+                                           submitTime: Long,
+                                           limit: Int,
+                                           dispatcher: PlanDispatcher,
+                                           dataset: DatasetRef,
+                                           shard: Int,
+                                           filters: Seq[ColumnFilter],
+                                           chunkMethod: ChunkScanMethod,
+                                           schema: Option[String] = None,
+                                           colName: Option[String] = None) extends LeafExecPlan {
```

Comment:
I dont understand colName selection fully. Can this be documented clearly somewhere ?
* How is colName chosen?
* How do we override colName for downsampled data ?
* Can user override colName in logical plan? I wonder if schema name (__type__ tag) should be used in logical plan instead of col name? Or is both `__type__` and `__col__` needed?

Reply:
I added docs for this field.